spark/src/main/scala/org/apache/iceberg/spark/SparkTableUtil.scala

```diff
@@ -420,7 +420,7 @@ object SparkTableUtil {
   private def buildManifest(
       conf: SerializableConfiguration,
       spec: PartitionSpec,
-      basePath: String): Iterator[SparkDataFile] => Iterator[Manifest] = { files =>
+      basePath: String): Iterator[SparkDataFile] => Iterator[ManifestFile] = { files =>
```

Comment:
We need to use the real `ManifestFile` with all stats if we want to skip reading/writing manifests on the driver.

Reply:
As long as this works, I'm all for it. Can we do the same with `DataFile`?

Reply:
I think that should work. I've created #763. What about making `DataFile` and `ManifestFile` extend `Serializable`?

Reply:
I'm not sure whether it's a good idea to make an interface `Serializable`. I'll have to think about that one.